rune/libcontainer/configs/validate/validator.go

```diff
@@ -230,9 +230,14 @@ func (v *ConfigValidator) enclave(config *configs.Config) error {
 		return fmt.Errorf("Unsupported ra_type Configuration %v!\n", config.Enclave.RaType)
 	}
 
+	if config.Enclave.RaProductEnclave == sgx.Invalid_Environment {
```

Comment:
Don't use _Environment any more. Use ProductEnclave and and DebugEnclave instead. In addition, where os Invalid_Environment defined?

Reply:
> Don't use _Environment any more. Use ProductEnclave and and DebugEnclave instead. In addition, where os Invalid_Environment defined?

OK, but IAS spec uses Development Environment and Production Environment.  Invalid_Environment is defined in libenclave/attestation/sgx/attest.go in the first commit of this pr.

Reply:
The "signature type" is called in IAS but we still use "quote type" because aesmd calls it, and it sounds reasonable. So we have right to decide how to name it.

"Environment" sounds like confusing because it seems to imply an enclave can dynamically choose its running  environment. Actually debug or product attribute has been decided before signing enclave image.